addons/point_of_sale/static/tests/tours/helpers/ProductScreenTourMethods.js

```diff
@@ -163,6 +163,15 @@ class Do {
             },
         ];
     }
+    scan_barcode(barcode) {
+        return [
+            {
+                content: `trigger bus barcode_scanned '${barcode}'`,
+                trigger: '.pos', // The element here does not really matter as long as it is present
+                run: () => { odoo.__DEBUG__.services['web.core'].bus.trigger('barcode_scanned', barcode, $(".o_web_client")[0]); },
```

Comment:
Kindly try this.
```suggestion
                run: () => {
                    posmodel.env.barcode_reader.scan(barcode);
                },
```

And starting saas-16.3, you'll probably need to use:
```
posmodel.env.services.barcode_reader.scan(barcode)
```